pandas/_libs/tslibs/conversion.pyx

```diff
@@ -148,8 +145,8 @@ cdef int64_t cast_from_unit(
 
 
 cpdef inline (int64_t, int) precision_from_unit(
-        str unit,
-        NPY_DATETIMEUNIT out_reso=NPY_DATETIMEUNIT.NPY_FR_ns,
+    str unit,
+    NPY_DATETIMEUNIT out_reso=NPY_DATETIMEUNIT.NPY_FR_ns,
 ):
```

Comment:
won't this result in the same issue? `precision_from_unit` calls `get_conversion_factor`, which might error, but `precision_from_unit` isn't handling the exception

would one of the solutions from https://stackoverflow.com/questions/75544869/propagate-error-in-function-which-returns-tuple-of-int/75548002#75548002 work for that?

Reply:
The fix is that this moves the call to precision_from_unit down to after the Y/M cases are handled in cast_from_unit

Reply:
looks like it's still failing though
```
FAILED pandas/tests/indexes/period/test_indexing.py::TestGetIndexer::test_get_indexer2 - pytest.PytestUnraisableExceptionWarning: Exception ignored in: 'pandas._libs.tslibs.conversion.precision_from_unit'
```

Reply:
huh, not sure how i missed that locally.  will troubleshoot, thanks.